### PR TITLE
Setup CodeQL for SAST

### PR DESCRIPTION
**Pull request by eyitemi-paystack**
Setup CodeQL for SAST

Commit to add CodeQL workflow for Static App Security Testing.

Description
-----------
Commit to add CodeQL workflow for Static App Security Scanning.

Motivation and Context
----------------------
This commit allows us to run Github CodeQl as Github Actions and use CodeQL as a Static Application Security Testing tool to identify security vulnerabilities in code before it gets merged to `master` and deployments made.

Acceptance Criteria
-------------------

Screenshots (if appropriate)
----------------------------

Types of changes
----------------

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist
---------

- [x] My pull request addresses exactly one patch/feature.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful. (Please rebase, removing useless commits if not)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [ ] My change requires change(s) to the environment.
      
- [ ] My change requires change(s) to the database.
      
- [ ] Other information.

Fixes #0 (Enter the number for the issue this fixes. If you have not yet created an issue, please do so now or delete this line if you are only submitting a patch)

## Changes made by this pull request
- 
- 
-

### CODE REVIEW

**Comment by eyitemi-paystack** [comment]
@ahmed-paystack @Peter-John-paystack please help review :pray::pray:

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
```
Comment (Peter-John-paystack) [inline .github/workflows/codeql-analysis.yml:35]: @eyitemi-paystack  So the problem with CodeQL is that its not scanning our files correctly as we use ObjC and Swift. I have spoke to Jocelyn about this and she said if we are using Sonar to scan this repo which we are already for security vulnerabilities then it should be fine. Ill confirm with her and maybe if you can also check with her 😊

Comment (Peter-John-paystack) [reply .github/workflows/codeql-analysis.yml:35]: @eyitemi-paystack  this is the link to Sonar Cloud for this repo for context, https://sonarcloud.io/summary/new_code?id=PaystackHQ_paystack-ios

Comment (Peter-John-paystack) [reply .github/workflows/codeql-analysis.yml:35]: @eyitemi-paystack Awesome, thats what I spoke to Jocelyn about so if the security team is happy, we will be using Sonar for all mobile repos. Can we then close this PR?  cc: @ahmed-paystack

Comment (ahmed-paystack) [reply .github/workflows/codeql-analysis.yml:35]: Yes, please. Let's close this PR.

**Pull request closed without merging**